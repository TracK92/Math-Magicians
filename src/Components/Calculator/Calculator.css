
.calculator {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  width: 450px;
  height: 400px;
  margin: 0 20px 10px;
  gap: 1px;
}

.logo_sect img {
  height: 400px;
  max-width: 80%;
  display: flex;
  justify-content: center;
  margin: auto;
  align-items: center;
}

.calculator > * {
  padding: 25px;
  font-size: 1.5rem;
  border: 1px white solid;
  font-weight: bold;
}

.zero {
  grid-column: 1/ span 2;
}

main {
  display: flex;
  justify-content: space-between;
}

.input {
  grid-column: 1/ span 4;
  background-color: gray;
  text-align: right;
  justify-self: end;
  font-size: 2rem;
  width: 100%;
  padding: 15px;
  font-weight: bold;
  color: white;
  height: 80px;
}

.symbol {
  background-color: rgb(245, 140, 42);
}

button {
  cursor: pointer;
}

h2 {
  margin-top: 10px;
  margin-left: 20px;
  letter-spacing: 0.05rem;
}

@media screen and (max-width: 750px) {
  .calculator > * {
    padding: 15px;
  }

  .calculator {
    width: 350px;
  }

  main {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  h2 {
    width: 100%;
    text-align: center;
  }
}
