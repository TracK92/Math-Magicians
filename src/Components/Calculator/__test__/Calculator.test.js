import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../Calculator';

test('should render calculator', () => {
  const tree = renderer.create(<Router><Calculator /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
