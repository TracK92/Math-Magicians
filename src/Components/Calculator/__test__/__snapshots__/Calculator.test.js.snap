// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render calculator 1`] = `
<main
  data-testid="calculator-1"
>
  <section>
    <h2>
      Let\`s do some math!
    </h2>
  </section>
  <section
    className="calculator"
  >
    <p
      className="input"
    >
      0
    </p>
    <button
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="symbol"
      id="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="symbol"
      id="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="symbol"
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      id="2"
      onClick={[Function]}
      type="button"
    >
       2 
    </button>
    <button
      id="3"
      onClick={[Function]}
      type="button"
    >
       3 
    </button>
    <button
      className="symbol"
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="symbol"
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </section>
</main>
`;
